app/api/common.py

```diff
@@ -0,0 +1,31 @@
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import exceptions
+
+from app import models
+
+
+def get_and_check_project(request, project_pk, perms=('view_project',)):
+    '''
+    Retrieves a project and raises an exeption if the current user
+    has no access to it.
+    '''
+    try:
+        project = models.Project.objects.get(pk=project_pk, deleting=False)
+        for perm in perms:
+            if not request.user.has_perm(perm, project): raise ObjectDoesNotExist()
```

Comment:
Is there a "no access" in Django Core, or would that be layered on by some authorization framework? It seems like it might be appropriate to make that distinction here even if it's immediately caught and translated to a 404 (keeping 403s out of the mix makes sense to avoid leaking object existence).